pkg/fanal/analyzer/language/python/pip/pip.go

```diff
@@ -41,3 +96,123 @@ func (a pipLibraryAnalyzer) Type() analyzer.Type {
 func (a pipLibraryAnalyzer) Version() int {
 	return version
 }
+
+// licenseList returns list of licenses found in METADATA files in the `site-packages` directory.
+func (a pipLibraryAnalyzer) licenseList() (map[string][]string, error) {
```

Comment:
If Python has a strict convention for package directory, it's more efficient to collect the license after we detect package names and versions, like [gomod](https://github.com/aquasecurity/trivy/blob/3eecfc6b6e1d6cef1497f7ff0044e676da159243/pkg/fanal/analyzer/language/golang/mod/mod.go#L151-L163), rather than collecting all licenses and keeping them in memory.

Reply:
hm... it make sense. I will update PR.

Reply:
Updated - https://github.com/aquasecurity/trivy/pull/6782/commits/beaeb6e2d525d1e4b106a90d161b0004e0975330